scrapy_po/middleware.py

```diff
@@ -20,9 +23,51 @@ class InjectionMiddleware:
     for this argument.
 
     XXX: should it really be a downloader middleware?
-    XXX: can this middleware allow to skip downloading
-    a page if it is not needed?
     """
+    def is_response_going_to_be_used(self, request, spider):
```

Comment:
👍 for extracting it, and not having the logic in process_request method. 

Could you please also write some tests for it? 

It seems this method doesn't need `self`, so you can move/refactor it to separate function(s), and test in isolation - e.g. detection of cases like `def parse(self, resp, page: ProductPage, response: DummyResponse):`. It looks feasible to write tests involving page objects as well.

Reply:
I've introduced new test cases. Feedback is appreciated 😄